root/static/scripts/relationship-editor/utility/prepareHtmlFormSubmission.js

```diff
@@ -121,15 +124,18 @@ function pushRelationshipHiddenInputs(
   pushInput(relPrefix, 'period.end_date.year', String(endDate?.year ?? ''));
   pushInput(relPrefix, 'period.end_date.month', String(endDate?.month ?? ''));
   pushInput(relPrefix, 'period.end_date.day', String(endDate?.day ?? ''));
-  pushInput(relPrefix, 'period.ended', relationship.ended ? '1' : '0');
+  pushInput(
+    relPrefix,
+    'period.ended',
+    hasDates && relationship.ended ? '1' : '0',
+  );
   pushInput(relPrefix, 'backward', backward ? '1' : '0');
 
   const linkTypeId = relationship.linkTypeID;
   if (linkTypeId != null) {
     pushInput(relPrefix, 'link_type_id', String(linkTypeId));
```

Comment:
Just a tiny cleanup suggestion, since we now have the `linkType` above.
```suggestion
  if (linkType != null) {
    pushInput(relPrefix, 'link_type_id', String(linkType.id));
```

Reply:
`getRelationshipLinkType` can return `null` even if `linkTypeId != null` which is why I didn't dare to make this change. Are you sure they're functionally equivalent?

Reply:
It's not equivalent if the link type isn't loaded for some reason, but it should always be loaded. If it wasn't, the existing line below would've already thrown an exception:

```JS
    const linkType = linkedEntities.link_type[linkTypeId];
    if (linkType.orderable_direction !== 0) {
```